InternetStoreCHIReloaded/DAL/Repositories/UsersRepository.cs

```diff
@@ -0,0 +1,220 @@
+                    dbResponse.Message = "Product entry was removed successfuly!";
+                }
+                else
+                {
+                    dbResponse.IsSuccessful = false;
+                    dbResponse.Message = "Nothing has changed.";
+                }
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Product was not found in cart";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse<int>> GetQuantityOfProductInCartAsync(int userId, int productId)
+        {
+            var user = await GetUserWithCartAsync(userId);
+            var product = user.UserCart.CartItems.Where(p => p.ProductId == productId).FirstOrDefault();
+
+            var dbResponse = new DbResponse<int>();
+
+            if (product != null)
+            {
+                dbResponse.IsSuccessful = true;
+                dbResponse.Message = "Product quantity retrieval success!";
+                dbResponse.Data = product.Quantity;
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Product was not found in cart";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse> MakeAnOrder(int userId)
+        {
+            var user = await GetUserWithCartAsync(userId);
+
+            DbResponse dbResponse = new DbResponse();
+
+            if (!user.UserCart.CartItems.Any())// if no items in cart
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Cart is empty.";
+                return dbResponse;
+            }
+
+            // getting products only because can't include the product entities in user cart
+            var productIds = user.UserCart.CartItems.Select(i => i.Id).ToList();
+
+            var productsWithQuantity = await _dbcontext.ProductsWithQuantity.Where(p => productIds.Contains(p.Id)).Include(p => p.Product).ToListAsync();
+
+            DateTime orderCreationdate = DateTime.UtcNow;
+
+            var orderedProducts = productsWithQuantity.Select(pwq => new OrderedProductEntity()
+            {
+                OrderedDate = orderCreationdate,
+                OrderedPrice = pwq.Product.Price,
+                Product = pwq.Product,
+                Quantity = user.UserCart.CartItems.Where(i => i.ProductId == pwq.ProductId).First().Quantity
+            }).ToList();
+
+            OrderEntity newOrder = new OrderEntity()
+            {
+                CreationDate = orderCreationdate,
+                OrderItems = orderedProducts,
+                TotalSum = orderedProducts.Sum(p => p.OrderedPrice)
+            };
+
+            user.UserOrders.Add(newOrder);
+
+            _dbcontext.ProductsWithQuantity.RemoveRange(productsWithQuantity);
+
+            user.UserCart.CartItems.Clear();
```

Comment:
Are we going to clear CartItems and ProductsWithQuantity? Are they same?

Reply:
done